market-generator/src/main/java/com/market/banica/generator/configuration/MarketConfigurationImpl.java

```diff
@@ -0,0 +1,167 @@
+                                     double priceLow, double priceHigh, double priceStep,
+                                     int periodLow, int periodHigh, int periodStep) {
+
+        try {
+            this.lock.writeLock().lock();
+
+            origin = origin.toLowerCase();
+            good = good.toLowerCase();
+
+            if (doesGoodExist(String.format(ORIGIN_GOOD_PATTERN, origin, good))) {
+                throw new IllegalArgumentException(String.format("A good with name %s from %s already exists",
+                        good.toUpperCase(), origin.toUpperCase()));
+            }
+
+            Properties properties = new LinkedProperties();
+
+            GoodSpecification goodSpec = new GoodSpecification(good, quantityLow, quantityHigh, quantityStep,
+                    priceLow, priceHigh, priceStep, periodLow, periodHigh, periodStep);
+
+            this.modifyProperties(goodSpec.generateProperties(origin), properties,
+                    properties::setProperty, this.file, false);
+            this.goods.put(String.format(ORIGIN_GOOD_PATTERN, origin, good), goodSpec);
+        } finally {
+            this.lock.writeLock().unlock();
+        }
+
+
+//        System.out.println("Adding"); TODO: replace writing in console with Logging
+    }
+
+    @Override
+    @ManagedOperation
+    public void removeGoodSpecification(String origin, String good) {
+        try {
+            this.lock.writeLock().lock();
+
+            origin = origin.toLowerCase();
+            good = good.toLowerCase();
+
+            if (!doesGoodExist(String.format(ORIGIN_GOOD_PATTERN, origin, good))) {
+                throw new NotFoundException(String.format("A good with name %s from %s does " +
+                        "not exist and it cannot be removed", good.toUpperCase(), origin.toUpperCase()));
+            }
+            Properties properties = new LinkedProperties();
+            Map<String, String> stringStringMap = this.goods.remove(String.format(ORIGIN_GOOD_PATTERN, origin, good)).generateProperties(origin);
+            this.modifyProperties(stringStringMap, properties,
+                    (k, v) -> properties.remove(k), this.file, false);
+
+            this.goods.remove(String.format(ORIGIN_GOOD_PATTERN, origin, good));
+
+//        System.out.println("Deleting"); TODO: replace writing in console with Logging
+        } finally {
+            this.lock.writeLock().unlock();
+        }
+    }
+
+    @Override
+    @ManagedOperation
+    public void updateGoodSpecification(String origin, String good, long quantityLow, long quantityHigh,
+                                        long quantityStep,
+                                        double priceLow, double priceHigh, double priceStep,
+                                        int periodLow, int periodHigh, int periodStep) {
+
+        try {
+            this.lock.writeLock().lock();
+
+            origin = origin.toLowerCase();
```

Comment:
there is a Sonar rule for this, right now you are not using Sonar to check your code but in future, you are going to have problems if using it. The rule is String operations should not rely on the default system locale. You could use it for example like: toLowerCase(Locale.getDefault())